src/naive_bayes/mod.rs

```diff
@@ -0,0 +1,239 @@
+use crate::error::Failed;
+use crate::linalg::BaseVector;
+use crate::linalg::Matrix;
+use crate::math::num::RealNumber;
+use std::marker::PhantomData;
+
+/// Distribution used in the Naive Bayes classifier.
+pub trait NBDistribution<T: RealNumber, M: Matrix<T>> {
+    /// Prior of class k
+    fn prior(&self, k: T) -> T;
+
+    /// Conditional probability of feature j given class k
```

Comment:
I think doc string for this function should be
```
Conditional probability of sample j given class k
```
because this function returns product of conditional probabilities, right?
![image](https://user-images.githubusercontent.com/1102497/98488643-40971980-21df-11eb-98d9-7ec6f8c937f1.png)